bee-tangle/src/vertices.rs

```diff
@@ -54,36 +54,58 @@ impl Vertices {
         unsafe { self.tables.get_unchecked(index) }
     }
 
-    pub(crate) async fn get(&self, message_id: &MessageId) -> Option<RwLockReadGuard<'_, Vertex>> {
+    pub(crate) fn get_map<T>(&self, message_id: &MessageId, f: impl FnOnce(&Vertex) -> T) -> Option<T> {
         let hash = self.make_hash(message_id);
-        let table = self.get_table(hash).read().await;
 
-        RwLockReadGuard::try_map(table, |table| match table.get(hash, equivalent_id(message_id)) {
-            Some((_, v)) => Some(v),
-            None => None,
-        })
-        .ok()
+        let guard = self.get_table(hash).read();
+        let output = guard.get(hash, equivalent_id(message_id)).map(|(_, v)| f(v));
+        drop(guard);
```

Comment:
Does the `drop(guard)` influence the performance compared to just dropping the lock when the `guard` goes out of scope? If not, I'm in favor of keeping omitting the `drop()` as it can lead to errors. I might even go as far as to provide a reason in a comment for why manually dropping a lock is safe, similar to the comments that we write for `.unwrap()`.